duckduckgo/Android
# Feature/send fire pixel on app launch

Task/Issue URL: https://app.asana.com/0/414730916066338/838465040462722
Tech Design URL: 
CC: 

**Description**:
Previously, there were ways in which the pixels for performing a 🔥could have been lost. This might have happened if during the fire animation, the user killed the app. When they next relaunched the app, we lost information about the 🔥. Their data would still have been safely deleted, as expected, but the pixel would have been lost.

**Steps to test this PR**:
1. From a fresh install, ensure the app launch pixel occurs.
1. Perform a 🔥 and allow the process to restart without intervention. Verify that 
    1. the fire pixel is sent
    1. the app launch pixel is **not sent**
1. Perform a 🔥, this time hit the home button during the fire animation. Wait 15s before relaunching the app from recents
    1. Verify the app doesn't re-show itself automatically
    1. Verify the fire pixel is sent
    1. Verify the app launch pixel **is** sent
1. Perform a 🔥, this time hit the home button during the fire animation. Kill the app entirely. Wait 15s before launching from the normal launcher icon
    1. Verify the app doesn't re-show itself automatically
    1. Verify the fire pixel is sent
    1. Verify the app launch pixel **is** sent

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
